packages/backend/src/libp2p/processInChunks.ts

```diff
@@ -0,0 +1,48 @@
+import logger from '../logger'
+const log = logger('dialer')
+
+const DEFAULT_CHUNK_SIZE = 10
+
+export class ProcessInChunks<T> {
+  private isActive: boolean
+  private data: T[]
+  private chunkSize: number
+  private processItem: (arg: any) => Promise<any>
+
+  constructor(data: T[], processItem: (arg: T) => Promise<any>, chunkSize: number = DEFAULT_CHUNK_SIZE) {
+    this.data = data
+    this.processItem = processItem
+    this.chunkSize = chunkSize
+    this.isActive = true
+  }
+
+  async processOneItem() {
+    if (!this.isActive) return
+    const toProcess = this.data.shift()
+    if (toProcess) {
+      try {
+        await this.processItem(toProcess)
+      } catch (e) {
+        log(`Processing ${toProcess} failed, message:`, e.message)
+      } finally {
+        process.nextTick(async () => {
+          await this.processOneItem()
+        })
+      }
+    }
+  }
+
+  async process() {
+    log(`Processing ${Math.min(this.chunkSize, this.data.length)} items`)
+    for (let i = 0; i < this.chunkSize; i++) {
+      await this.processOneItem()
```

Comment:
you shouldn't `await` here right?
if you await you don't do stuff concurrently, but one by one

Reply:
True, I added this mindlessly while fixing linter's `no-floating-promises`